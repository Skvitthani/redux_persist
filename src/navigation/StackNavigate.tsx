import React from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

// Screen
import GetUserData from '../screens/GetUserData';
import AddUserDataScreen from '../screens/AddUserDataScreen';

// Type
import {DealerNavigNavigationType} from '.';

const Stack = createNativeStackNavigator<DealerNavigNavigationType>();

const StackNavigate = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator screenOptions={{headerShown: false}}>
        <Stack.Screen name="AddUserDataScreen" component={AddUserDataScreen} />
        <Stack.Screen name="GetUserData" component={GetUserData} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default StackNavigate;
