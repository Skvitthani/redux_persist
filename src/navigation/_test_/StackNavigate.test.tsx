import {render} from '@testing-library/react-native';
import React from 'react';
import StackNavigate from '../StackNavigate';

jest.mock('@react-navigation/native-stack', () => ({
  createNativeStackNavigator: () => ({
    Navigator: 'mocked-Navigator',
    Screen: 'mocked-Screen',
  }),
}));

jest.mock('react-redux', () => ({
  Provider: jest.fn(),
}));

const component = <StackNavigate />;

describe('StackNavigate', () => {
  it('renders correctly', () => {
    const container = render(component);

    expect(container.toJSON()).toMatchSnapshot();
  });
});
