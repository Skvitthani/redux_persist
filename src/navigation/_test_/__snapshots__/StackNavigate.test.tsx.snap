// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackNavigate renders correctly 1`] = `
<mocked-Navigator
  screenOptions={
    {
      "headerShown": false,
    }
  }
>
  <mocked-Screen
    component={[Function]}
    name="AddUserDataScreen"
  />
  <mocked-Screen
    component={[Function]}
    name="GetUserData"
  />
</mocked-Navigator>
`;
